library/src/scala/tasty/Reflection.scala

```diff
@@ -488,7 +488,28 @@ class Reflection(private[scala] val internal: CompilerInterface) { self =>
       /** Shows the tree as fully typed source code */
       def showWith(syntaxHighlight: SyntaxHighlight)(using ctx: Context): String =
         new SourceCodePrinter[self.type](self)(syntaxHighlight).showTree(tree)
+
+      /** Does this tree represent a valid expression? */
+      def isExpr(using ctx: Context): Boolean =
+        tree match
+          case tree: Term =>
+            tree.tpe.widen match
+              case _: MethodType | _: PolyType => false
+              case _ => true
+          case _ => false
+
```

Comment:
The `isExpr` will cause confusion for meta-programmers: what's the relation between `expression` and `Expr[T]`? What is the relationship between `expression` and `Term`?

Reply:
The is `isExpr` indicates that this term is a valid `Expr[Any]`, therefore `asExprOf` will succeed. The names are intentionally similar as the are about the same concept. Docs can improve.

Reply:
Maybe make the method private?

Reply:
But users are intended to use it